packages/calendar/src/Calendar.ts

```diff
@@ -0,0 +1,511 @@
+                narrow: this.formatDate(date, { weekday: 'narrow' }),
+                long: this.formatDate(date, { weekday: 'long' }),
+            };
+        });
+    }
+
+    /**
+     * Defines the initial date that will be used to render the calendar, if no specific date is provided
+     */
+    private setInitialCalendarDate(): void {
+        this.currentDate = this.today;
+    }
+
+    /**
+     * If a date is received by the component via property, it uses that date as the current date to render the calendar
+     */
+    private setCurrentCalendarDate(): void {
+        if (!this.selectedDate) {
+            return;
+        }
+
+        this.selectedDate = new Date(this.selectedDate);
+
+        if (!this.isValidDate(this.selectedDate)) {
+            this.selectedDate = undefined;
+            return;
+        }
+
+        this.currentDate = this.toCalendarDate(this.selectedDate);
+    }
+
+    /**
+     * Sets the minimum allowed date a user can select by converting a `Date` object to `CalendarDate`, which is the
+     * type of object used internally by the class
+     */
+    private setMinCalendarDate(): void {
+        if (!this.min) {
+            return;
+        }
+
+        this.min = new Date(this.min);
+
+        if (!this.isValidDate(this.min)) {
+            this.min = undefined;
+            return;
+        }
+
+        this.minDate = this.toCalendarDate(this.min);
+    }
+
+    /**
+     * Sets the maximum allowed date a user can select by converting a `Date` object to `CalendarDate`, which is the
+     * type of object used internally by the class
+     */
+    private setMaxCalendarDate(): void {
+        if (!this.max) {
+            return;
+        }
+
+        this.max = new Date(this.max);
+
+        if (!this.isValidDate(this.max)) {
+            this.max = undefined;
+            return;
+        }
+
+        this.maxDate = this.toCalendarDate(this.max);
+    }
+
+    /**
+     * Returns an array with all days of the week corresponding to the given index, starting with the first day of the
+     * week according to the locale
+     *
+     * @param weekIndex - The index of the week
+     */
+    private getDatesInWeek(weekIndex: number): CalendarDate[] {
+        const dates: CalendarDate[] = [];
+
+        let date = startOfWeek(
+            startOfMonth(this.currentDate).add({ weeks: weekIndex }),
+            this.locale
+        );
+
+        while (dates.length < daysInWeek) {
+            dates.push(date);
+
+            const nextDate = date.add({ days: 1 });
+
+            // If the next day is the same, we have hit the end of the calendar system
+            if (isSameDay(date, nextDate)) {
+                break;
+            }
+
+            date = nextDate;
+        }
+
+        return dates;
+    }
+
+    /**
+     * Converts a `Date` object to a `CalendarDate`
+     *
+     * @param date - `Date` object to be converted
+     */
+    private toCalendarDate(date: Date): CalendarDate {
```

Comment:
From here down, are there any other locations you might want to use similar methods that would make sense for them to be utility methods rather than class methods?